Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy+Error.swift

```diff
@@ -231,6 +247,20 @@ extension PathHierarchy.Error {
         }
         return "Replace \(from.singleQuoted) with \(to.singleQuoted)"
     }
+
+    /// Find an overload group node from an array of disambiguation candidates. Only
+    /// return the overload group if all the remaining candidates are disfavored for link resolution.
+    /// - Parameters:
+    ///   - candidates: An array of candidate node and disambiguation tuples.
+    /// - Returns: The PathHierarchy node for the overload group symbol
+    func overloadGroupNodeFrom(candidates: [(node: PathHierarchy.Node, disambiguation: String)]) -> PathHierarchy.Node? {
+        guard FeatureFlags.current.isExperimentalOverloadedSymbolPresentationEnabled,
+            let favoredCandidate = candidates.singleMatch({ !$0.node.specialBehaviors.contains(.disfavorInLinkCollision) }),
+            favoredCandidate.node.symbol?.isOverloadGroup ?? false else {
```

Comment:
Looks like removing this condition (and reducing the guard to just the `singleMatch`) will solve David's concern and make this more robust.